src/main/java/org/opensearch/neuralsearch/search/query/HybridCollectorManager.java

```diff
@@ -60,27 +67,42 @@ public static CollectorManager createHybridCollectorManager(final SearchContext
         int numDocs = Math.min(searchContext.from() + searchContext.size(), totalNumDocs);
         int trackTotalHitsUpTo = searchContext.trackTotalHitsUpTo();
 
+        Weight filteringWeight = null;
+        // Check for post filter to create weight
+        if (Objects.nonNull(searchContext.parsedPostFilter())) {
+            Query filterQuery = searchContext.parsedPostFilter().query();
```

Comment:
we should check if query object isn't null

Reply:
This check would become redundant as when parsedPostFilter is set and if query is null then below line will handle that situation https://github.com/opensearch-project/OpenSearch/blob/77a119319c14b0c1fa45dc85a618bf2429576b1e/server/src/main/java/org/opensearch/search/SearchService.java#L1255

Reply:
To me we always need to do basic validation for cases like this, this searchContext is constructed not even in the plugin, and logic that you've referenced can change anytime, or new logic can be added. Let's not rely on good intensions in other components